### ISSUE TITLE
Integration testing environment

### ISSUE THREAD

**Issue by netrome** [stacks-network/sbtc#53]
When developing applications on sBTC it is important to be able to integration test them. We need to make it easy to spin up and control a bitcoin regtest node and a stacks node connecting to the bitcoin node. The nodes should be programmatically controllable (mine new blocks etc), and it should be easy to interface with them from the code the same way one would interface with live nodes.

### Definition of done
We have a module that supports programmatic control of a bitcoin regtest and a stacks devnet node, allowing for smooth and efficient integration testing.

**Comment by AshtonStephens** [comment]
The plan at the moment is to use the clarinet cli https://github.com/hirosystems/clarinet command `clarinet integrate`

Core requirements:
- run clarinet integrate
- deploy sbtc contracts
- query sbtc nodes for data

**Comment by netrome** [comment]
Clarinet integrate docs: https://docs.hiro.so/clarinet/feature-guides/clarinet-integrate

**Comment by AshtonStephens** [comment]
Update: We know how to run `clarinet integrate` and are now looking into how we can use the `orchestrator` functionality from the `stacks-devnet-js` package via the cli. Note that we're planning to wrap around the clarinet integrate CLI so that that code isn't under the GPL license. 

Example of the orchestrator: https://github.com/hirosystems/stacks-e2e-testing/blob/main/tests/integration/epoch-2-2/wrap-trait-workaround.spec.ts#L59

**Comment by friedger** [comment]
Here is an example how to test stacking using the orchestrator from `stacks-devnet-js`: https://github.com/friedger/clarity-stacking-pools/blob/main/integration/stacking-self-service.test.ts and https://github.com/friedger/clarity-stacking-pools/blob/main/integration/stacking-1-cycle.test.ts

**Comment by CAGS295** [comment]
My feeling is that 'clarinet integrate/deployments' is superior as a node orchestrator, than DevnetJS's. To me, devnet JS is too slow and the internals are hard to follow given the current implementation. Although I miss the event-drivenness that devnetJS provides. e.g. mining and waiting for blocks. I hope that chainhooks (once I dig into it) could be the missing link to make an integration suite built on top of commodity clarinet tools and stacksJS.

**Comment by CAGS295** [comment]
I can see to immediate alternatives.
1. Craft integration tests in clarity. Spin a devnet environment with clarinet integrate. Use deployments to execute your tests written in clarity.
2. Use DevnetJS to spin a devnet environment. Use the provided orchestrator and @stacks/network to implement tests in plain typescript. Use something like jest or vitest to run your tests.

Both are viable options. None is perfect atm.

**Comment by AshtonStephens** [comment]
Closing because this ticket is **stale**. The work, as represented here, is not going to happen. Another ticket for smoother integration testing will likely take it's place. We will reference this ticket there.

**Comment by AshtonStephens** [comment]
@CAGS295 is working on this - I don't think he has cut the tickets yet but I'll ping him. He's going to design the system and then break it up into components and then execute.
